drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -934,15 +910,15 @@ local function level_attr_handler(driver, device, ib, response)
 end
 
 local function hue_attr_handler(driver, device, ib, response)
-  if ignore_initial_color_read(device, color_mode_attr_bits.HUE) or ib.data.value == nil then
+  if device:get_field(COLOR_MODE) == X_Y_COLOR_MODE  or ib.data.value == nil then
```

Comment:
might be better to do:
```suggestion
  if device:get_field(COLOR_MODE) ~= HUE_SAT_COLOR_MODE  or ib.data.value == nil then
```

Reply:
and the same with the other handlers

Reply:
The problem with that is I have seen a Zemismart bulb report that its color mode was `COLOR_TEMPERATURE` in its subscription report following a device power cycle, even though it was previously `CURRENT_HUE_SAT` and the bulb was clearly still emitting a color. It didn't seem to report the proper color mode until a color change command was sent to the device and forced it to update. So with the current check, this problem would be avoided at least.